.github/workflows/build.yml

```diff
@@ -71,9 +71,21 @@ jobs:
           ssh-key: ${{ secrets.SSH_PRIVATE_KEY }}
           path: strains
 
-      - name: Install necessary dependencies
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.11
+
+      - name: Setup Virtual Environment
+        run: |
+          python -m venv venv
+          . venv/bin/activate
+          echo PATH=$PATH >> $GITHUB_ENV
```

Comment:
Why do we do this?

Reply:
I explained it better in this commit: https://github.com/eduNEXT/picasso/pull/30/commits/99d4ca823100547353e488ee5ab64faa798da95d

Reply:
So the ubuntu 24.04 migration was rolled out, that's why the workflow is currently working fine: https://github.com/actions/runner-images/issues/10781#issuecomment-2417343919

Reply:
The combination of setup-python with Python 3.xx and pinning the Ubuntu image would be a good enough approach to solve this in case it happens again. However, as I mentioned, this has stopped happening since the rollout. Still, this raised a really good question about pinning GH environment requirements, which we can address now.

Reply:
Thanks for the context @mariajgrimaldi. I mean, as far as I understand the environment we only need it because of that error. After it's fixed we can remove that step from the workflow? Or we will leave it like that from now on.

Reply:
This step is one of the possible workarounds that fixed this issue (according to [this](https://github.com/actions/runner-images/issues/10781#issuecomment-2416762359) comment). But I'm going with using a fixed Python version, so we can manually control updates from now on: https://github.com/eduNEXT/picasso/pull/30/commits/e5639fa36180419cdc36c8eddd473493906c12ff

That's just one of the reasons why I'm pinning Python though, another is that not all of our tutor plugins (e.g. codejail) support Python 3.12 so builds fail with that version.

Reply:
@mariajgrimaldi, thank you for all the info and for catching this. Since the GitHub runner error isn't something that should recur, what do you think about keeping the runner at the latest?

I agree with using a fixed Python version —do you think there might be cases where someone using this reusable workflow would want to use a different Python version than the one we're pinning? I can't think of any specific examples at the moment

Reply:
@magajh: It shouldn't happen, but it could happen again. This time, it was Python, but other dependencies (docker, compose, etc.) could fail as well. I'm okay going forward with using the latest only if we monitor these updates closely. We can put a pin on this and use latest for now:

As for the Python version, the only case that comes to mind is that we move to Python 3.12 by default but some other organization plugins only support Python 3.11, then they'd want to use another default.

Reply:
@mariajgrimaldi hello again. I've been thinking and weighing the pros and cons, and I agree that using a fixed version for the runner will give us better control over updates and help prevent unexpected problems.

About the Python version, how about in the near future we also allow users to specify a different Python version if they need to? If they don't provide a version, we can default to the one we've pinned, so we can still offer some flexibility while maintaining stability

Reply:
Sounds good! https://github.com/eduNEXT/picasso/pull/30/commits/58d440ca44a949b4e8382f3c8ecfbd4af8d5877f